### PR TITLE
WIP: fixing broken tests on cygwin

### PR DESCRIPTION
**Pull request by garbas**
below changes make test on cygwin pass, but this PR should not be merged as it is currently.

couple of things are currently not working with nix+cygwin:
* [ ] when --enable-gc is enabled many tests fail
* [ ] $shell variable is empty which fails the tests, need to figure out the reason why

[Review on Reviewable](https://reviewable.io/reviews/nixos/nix/541)

### CODE REVIEW

**Comment by garbas** [comment]
maybe makes sense to add that inside cygwin $SHELL is set to "/bin/bash"

File: release.nix [hunk 1]
```diff
@@ -6,7 +6,7 @@ let
 
   pkgs = import <nixpkgs> {};
 
-  systems = [ "x86_64-linux" "i686-linux" "x86_64-darwin" /* "x86_64-freebsd" "i686-freebsd" */ ];
+  systems = [ "x86_64-linux" "i686-linux" "x86_64-darwin" "i686-cygwin" "x86_64-cygwin" /* "x86_64-freebsd" "i686-freebsd" */ ];
```
Comment (edolstra) [inline release.nix]: This won't work because we don't have Cygwin machines in the build farm...

Comment (copumpkin) [inline release.nix]: Can we contribute money somewhere to add one? :smile:

Comment (garbas) [inline release.nix]: np, i will extend release.nix to be able to pass extra systems to built on, for now.

Comment (garbas) [inline release.nix]: @edolstra i've looked into release.nix again. ``systems`` only lists platforms that nix can be built on. later in the release.nix there is a release attribute  (line 231) and all platforms that hydra will build are listed there. please correct me if i am wrong.

Comment (edolstra) [inline release.nix]: The release attribute only determines the release job (http://hydra.nixos.org/job/nix/master/release). But this line:
```
    build = pkgs.lib.genAttrs systems (system:
```
causes Hydra to instantiate a job `build.x86_64-cygwin` that it won't be able to build.

Comment (garbas) [inline release.nix]: i hope new commit fixes this properly.

**Comment by garbas** [comment]
i messed up something on my fork. will create new PR once i resurrect it

**Pull request closed without merging**